samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -69,6 +69,115 @@ def full_address(self) -> str:
         return self.address
 
 
+def _build_module(
+    module_name: Optional[str],
+    module_configuration: Dict,
+    input_variables: Dict[str, Expression],
+    parent_module_address: Optional[str],
+) -> TFModule:
+    module = TFModule(None, None, {}, [], {}, {})
+
+    module.full_address = _build_module_full_address(module_name, parent_module_address)
+    module.variables = _build_module_variables_from_configuration(module_configuration, input_variables)
+    module.resources = _build_module_resources_from_configuration(module_configuration, module)
+    module.outputs = _build_module_outputs_from_configuration(module_configuration)
+    module.child_modules = _build_child_modules_from_configuration(module_configuration, module)
+
+    return module
+
+
+def _build_module_full_address(module_name: Optional[str], parent_module_address: Optional[str]) -> Optional[str]:
+    full_address = None
+    if module_name:
+        full_address = f"module.{module_name}"
+        if parent_module_address:
+            full_address = f"{parent_module_address}.{full_address}"
+
+    return full_address
+
+
+def _build_module_variables_from_configuration(
+    module_configuration: Dict, input_variables: Dict[str, Expression]
+) -> Dict[str, Expression]:
+    module_variables: Dict[str, Expression] = {}
+
+    default_variables = module_configuration.get("variables", {})
+    for variable_name, variable_value in default_variables.items():
+        module_variables[variable_name] = ConstantValue(variable_value.get("default"))
+    module_variables.update(input_variables)
+
+    return module_variables
+
+
+def _build_module_resources_from_configuration(module_configuration: Dict, module: TFModule) -> List[TFResource]:
+    module_resources = []
+
+    config_resources = module_configuration.get("resources", [])
+    for config_resource in config_resources:
+        resource_attributes: Dict[str, Expression] = {}
+
+        expressions = config_resource.get("expressions", {})
+        print(expressions)
```

Comment:
You can replace the print with debugging messages.

Reply:
ah missed removing this one, was debugging some failing tests at the time

Reply:
done